### ISSUE TITLE
False Positive for registered recipients post uninstall of Signal (temporary?)

### ISSUE THREAD

**Issue by paxperscientiam** [signalapp/Signal-Android#9105]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Describe here the issue that you are experiencing.

### Steps to reproduce
- Install Signal on phone A and phone B
- Register phone number of each respective phone with Signal
- Initiate correspondence between phone A and phone B
- Uninstall Signal from phone B
- Send button on phone A for conversation with phone B erroneously indicates that message will be encrypted/received

### Device info

**Device:** Motorola MotoG3
**Android version:** 7.1.2
**Signal version:** 4.48.17

### Link to debug log
https://debuglogs.org/80880b8e325ff5e86807fbf31f39beb8498837ad1e054e3c8bd8cac9e6006d82

**Comment by greyson-signal** [comment]
The only way we find out about uninstalls is through FCM, which historically has been quite slow to notify. We can't perform some action upon app uninstall ourselves, because, well, the app is uninstalled. Feel free to have your friend unregister here: https://signal.org/textsecure/unregister/

**Comment by paxperscientiam** [comment]
Hi @greyson-signal, is it clear to you what happens when phone A sends a message to phone B post phone B uninstall of Signal? Does Signal on phone A attempt to send encrypted message that then gets "lost in the ether"?

As far as determining whether Signal has been uninstalled from a phone, couldn't some sort of "ping" be used to ascertain that? A heuristic approach could be employed to warn a Signal user that a potential recipient may longer have Signal installed.

In short, I don't think it makes sense to rely on someone being responsible enough to manually de-register. That said, it seems that Signal *does* eventually figure out that recipient no longer uses Signal (seems to have taken a few days).
